jetty-webapp/src/test/java/org/eclipse/jetty/webapp/ServletContextTmpAttributeTest.java

```diff
@@ -0,0 +1,176 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.webapp;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import javax.servlet.ServletContext;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class ServletContextTmpAttributeTest
+{
+
+    /**
+     * ServletContext.TEMPDIR has <code>null</code> value
+     * but webappContent#tempDirectory is created under <code>java.io.tmpdir</code>
+     */
+    @Test
+    public void attributeWithNullValue() throws Exception
+    {
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        WebAppContext webAppContext = new WebAppContext();
+        webAppContext.setAttribute(ServletContext.TEMPDIR, null);
+        webInfConfiguration.resolveTempDirectory(webAppContext);
+        File tmp = webAppContext.getTempDirectory();
+        assertThat("webAppContext temp directory parent is java.io.tmpdir",
```

Comment:
Add an assert for the expected WebAppContext.isPersistentTempDirectory() value to all tests.

Reply:
The value of `isPersistentTempDirectory()` will always be false, because we never change it from its default value. And then these tests are not testing the `WebAppContext` in a server so the tmpDir would never be deleted anyway. I'm getting a bunch of exceptions trying to start `WebAppContext` on its own.

The `isPersistentTempDirectory()` is tested more thoroughly in the tests that I wrote so we probably don't need to test it here as well.